lib/plugins/explosion.js

```diff
@@ -77,7 +77,7 @@ function inject (bot) {
     // The following modifiers are constant for the input targetEntity and doesnt depend
     // on the source position, so if the goal is to compare between positions they can be
     // ignored to save computations
-    if (!rawDamages) {
+    if (!rawDamages && targetEntity.attributes['generic.armor']) {
```

Comment:
instead of returning undefined maybe we throw here? or return a null?

Reply:
The main reason I added this change is so that it doesn't throw. I would say we just return null instead